<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ServerFilterDock</class>
 <widget class="QDockWidget" name="ServerFilterDock">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>204</width>
    <height>437</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Maximum" vsizetype="Maximum">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string>Server Filter</string>
  </property>
  <widget class="QWidget" name="dockWidgetContents">
   <layout class="QVBoxLayout" name="verticalLayout_2">
    <item>
     <layout class="QVBoxLayout" name="verticalLayout">
      <item>
       <widget class="QLabel" name="label">
        <property name="text">
         <string>Server Name:</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLineEdit" name="leServerName">
        <property name="minimumSize">
         <size>
          <width>150</width>
          <height>0</height>
         </size>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="label_2">
        <property name="text">
         <string>Max. Ping:</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QSpinBox" name="spinMaxPing">
        <property name="toolTip">
         <string>Set '0' to disable.</string>
        </property>
        <property name="correctionMode">
         <enum>QAbstractSpinBox::CorrectToNearestValue</enum>
        </property>
        <property name="maximum">
         <number>9999</number>
        </property>
        <property name="singleStep">
         <number>10</number>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="label_3">
        <property name="text">
         <string>WADs:</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLineEdit" name="leWads">
        <property name="toolTip">
         <string>Use ',' (a comma) to separate multiple wads.</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="label_5">
        <property name="text">
         <string>Exclude WADs:</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLineEdit" name="leExcludeWads">
        <property name="toolTip">
         <string>Servers with WADs on this list won't be displayed. Use ',' (a comma) to separate multiple wads.</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="label_4">
        <property name="text">
         <string>Game modes:</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="MultiComboBox" name="cboGameMode"/>
      </item>
      <item>
       <widget class="QLabel" name="label_6">
        <property name="text">
         <string>Exclude game modes:</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="MultiComboBox" name="cboExcludeGameMode">
        <property name="toolTip">
         <string>Selected game modes won't appear on the server list.</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QCheckBox" name="cbShowFull">
        <property name="text">
         <string>Show full servers</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QCheckBox" name="cbShowEmpty">
        <property name="text">
         <string>Show empty servers</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QCheckBox" name="cbShowOnlyValid">
        <property name="text">
         <string>Show only valid servers</string>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QPushButton" name="btnClear">
        <property name="text">
         <string>Clear</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <spacer name="verticalSpacer">
      <property name="orientation">
       <enum>Qt::Vertical</enum>
      </property>
      <property name="sizeHint" stdset="0">
       <size>
        <width>20</width>
        <height>40</height>
       </size>
      </property>
     </spacer>
    </item>
   </layout>
  </widget>
 </widget>
 <customwidgets>
  <customwidget>
   <class>MultiComboBox</class>
   <extends>QComboBox</extends>
   <header>gui/widgets/multicombobox.h</header>
   <slots>
    <signal>valueChanged()</signal>
   </slots>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>leServerName</tabstop>
  <tabstop>spinMaxPing</tabstop>
  <tabstop>leWads</tabstop>
  <tabstop>leExcludeWads</tabstop>
  <tabstop>cboGameMode</tabstop>
  <tabstop>cboExcludeGameMode</tabstop>
  <tabstop>cbShowFull</tabstop>
  <tabstop>cbShowEmpty</tabstop>
  <tabstop>cbShowOnlyValid</tabstop>
  <tabstop>btnClear</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>btnClear</sender>
   <signal>clicked()</signal>
   <receiver>ServerFilterDock</receiver>
   <slot>clear()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>102</x>
     <y>419</y>
    </hint>
    <hint type="destinationlabel">
     <x>101</x>
     <y>210</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>leServerName</sender>
   <signal>textChanged(QString)</signal>
   <receiver>ServerFilterDock</receiver>
   <slot>emitUpdated()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>47</x>
     <y>57</y>
    </hint>
    <hint type="destinationlabel">
     <x>38</x>
     <y>392</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spinMaxPing</sender>
   <signal>editingFinished()</signal>
   <receiver>ServerFilterDock</receiver>
   <slot>emitUpdated()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>99</x>
     <y>107</y>
    </hint>
    <hint type="destinationlabel">
     <x>61</x>
     <y>404</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>leWads</sender>
   <signal>textEdited(QString)</signal>
   <receiver>ServerFilterDock</receiver>
   <slot>emitUpdated()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>123</x>
     <y>155</y>
    </hint>
    <hint type="destinationlabel">
     <x>72</x>
     <y>393</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cbShowFull</sender>
   <signal>clicked()</signal>
   <receiver>ServerFilterDock</receiver>
   <slot>emitUpdated()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>73</x>
     <y>318</y>
    </hint>
    <hint type="destinationlabel">
     <x>159</x>
     <y>400</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cbShowEmpty</sender>
   <signal>clicked()</signal>
   <receiver>ServerFilterDock</receiver>
   <slot>emitUpdated()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>47</x>
     <y>341</y>
    </hint>
    <hint type="destinationlabel">
     <x>133</x>
     <y>411</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cbShowOnlyValid</sender>
   <signal>clicked()</signal>
   <receiver>ServerFilterDock</receiver>
   <slot>emitUpdated()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>52</x>
     <y>364</y>
    </hint>
    <hint type="destinationlabel">
     <x>138</x>
     <y>384</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cboGameMode</sender>
   <signal>valueChanged()</signal>
   <receiver>ServerFilterDock</receiver>
   <slot>emitUpdated()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>58</x>
     <y>239</y>
    </hint>
    <hint type="destinationlabel">
     <x>99</x>
     <y>417</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>ServerFilterDock</sender>
   <signal>visibilityChanged(bool)</signal>
   <receiver>leServerName</receiver>
   <slot>setFocus()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>101</x>
     <y>210</y>
    </hint>
    <hint type="destinationlabel">
     <x>101</x>
     <y>60</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>leExcludeWads</sender>
   <signal>textEdited(QString)</signal>
   <receiver>ServerFilterDock</receiver>
   <slot>emitUpdated()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>101</x>
     <y>195</y>
    </hint>
    <hint type="destinationlabel">
     <x>101</x>
     <y>218</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cboExcludeGameMode</sender>
   <signal>valueChanged()</signal>
   <receiver>ServerFilterDock</receiver>
   <slot>emitUpdated()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>101</x>
     <y>285</y>
    </hint>
    <hint type="destinationlabel">
     <x>101</x>
     <y>218</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>emitUpdated()</slot>
  <slot>clear()</slot>
  <slot>emitUpdated(QString)</slot>
 </slots>
</ui>
